### DISCUSSION TITLE
Provide support for dynamic links

### DISCUSSION THREAD

**Discussion by EchoEllet** [PrismLauncher/PrismLauncher#2297] [Ideas]
I know this might be out of scope for the launcher but a lot of modern apps already has app/dynamic links

It could me something like this:

```
minecraft://connect?type=server&ip=127.0.0.1
```

To open a instance and join a Minecraft server for example, I'm not sure how it done on Desktop but I already did it on Mobile apps and it's pretty easy to implement

You just have to register the link, get the parameters, call the existing functions you already have to do what the link says

**Comment by roke-julian-lockhart** [comment]
> https://github.com/orgs/PrismLauncher/discussions/2297#discussion-6488335

@ellet0, the URN schema shouldn't use an upstream trademark without that being afore-standardized. It'd be better to use `prism://connect?type=server&ip=127.0.0.1`, since URNs aren't designed to be utilized outside of their intended context (unless published in IETF RFCs) and thus can reasonably just be a single, common word (like "prism").

Otherwise, this is a genius idea - I especially like that you've chosen it to primarily utilize arguments, rather than a hierarchical order (like a binary and its flags, rather than subdommands). That's more versatile.

**Reply by EchoEllet** [reply]
I agree. The previous url was an example that is already available in Minecraft bedrock.

For now, you can use Prism launcher CLI:
https://prismlauncher.org/wiki/getting-started/command-line-interface/

Even though this is not designed to send links, it's what I was looking for.

The url needs more parameters to provide details for the instance path or folder name or the game version, etc...
